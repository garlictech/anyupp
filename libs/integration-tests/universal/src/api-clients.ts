import {
  awsConfig,
  getCrudSdkForIAM,
  getCrudSdkForUserPool,
} from '@bgap/crud-gql/api';
import {
  getAnyuppSdkForIAM,
  getAnyuppSdkForUserPool,
} from '@bgap/anyupp-gql/api';
import { Auth } from 'aws-amplify';
import { from } from 'rxjs';
import { map } from 'rxjs/operators';

Auth.configure({
  ...awsConfig,
  // See: https://github.com/aws-amplify/amplify-js/issues/6552#issuecomment-682259256
  authenticationFlowType: 'USER_PASSWORD_AUTH',
  aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
});

export const createAuthenticatedCrudSdk = (
  userName: string,
  password: string,
) => from(Auth.signIn(userName, password)).pipe(map(getCrudSdkForUserPool));

export const createAuthenticatedAnyuppSdk = (
  userName: string,
  password: string,
) => from(Auth.signIn(userName, password)).pipe(map(getAnyuppSdkForUserPool));

export const createIamCrudSdk = () =>
  getCrudSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );

export const createIamAnyuppSdk = () =>
  getAnyuppSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );
