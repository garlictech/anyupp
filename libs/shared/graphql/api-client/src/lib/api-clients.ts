import { awsConfig } from '@bgap/crud-gql/api';
import { config } from '@bgap/shared/config';
import { ICrudApiConfig } from '@bgap/shared/types';
import { GraphqlApiFp } from './graphql-api-fp';

const AWS_APPSYNC_CONFIG: ICrudApiConfig = {
  ...awsConfig,
  aws_appsync_apiKey: config.GraphqlApiKey,
  aws_appsync_graphqlEndpoint: config.GraphqlApiUrl,
};

export const appsyncGraphQlClient = GraphqlApiFp.createPublicClient(
  AWS_APPSYNC_CONFIG,
  console,
  true,
);

const AWS_AMPLIFY_CONFIG: ICrudApiConfig = {
  ...awsConfig,
};

export const amplifyGraphQlClient = GraphqlApiFp.createPublicClient(
  AWS_AMPLIFY_CONFIG,
  console,
  true,
);
