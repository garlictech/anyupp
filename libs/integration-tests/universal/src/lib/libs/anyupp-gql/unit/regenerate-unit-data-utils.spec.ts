import {
  getProductComponentMap,
  getProductComponentSetMap,
  getTimezoneForUnit,
  listUnitProductsForAUnit,
} from '@bgap/anyupp-gql/backend';
import { createIamCrudSdk } from '../../../../api-clients';

const TEST_NAME = 'REGEN_UTILS';

describe('RegenerateUnitData Utils tests', () => {
  const iamCrudSdk = createIamCrudSdk();

  describe('getProductComponentSetMap function', () => {
    it('should still emit an empty object in case there is NO data in the DB', done => {
      getProductComponentSetMap({
        crudSdk: iamCrudSdk,
      })('NOT_EXISTING_CHAIN_ID').subscribe({
        next(result) {
          expect(result).toEqual({});
          done();
        },
        error(err) {
          console.error(`${TEST_NAME}Test ERROR`, err);
        },
      });
    }, 15000);
  });

  describe('getProductComponentMap function', () => {
    it('should still emit an empty object in case there is NO data in the DB', done => {
      getProductComponentMap({
        crudSdk: iamCrudSdk,
      })('NOT_EXISTING_CHAIN_ID').subscribe({
        next(result) {
          expect(result).toEqual({});
          done();
        },
        error(err) {
          console.error(`${TEST_NAME}Test ERROR`, err);
        },
      });
    }, 15000);
  });

  describe('listUnitProductsForAUnit function', () => {
    it('should throw in case there are NO unitProducts in the DB', done => {
      listUnitProductsForAUnit({
        crudSdk: iamCrudSdk,
      })('NOT_EXISTING_UNIT_ID').subscribe({
        next() {
          console.error(`${TEST_NAME}Test ERROR`, 'SHOULD NOT SUCCEED');
        },
        error(err) {
          expect(err).toMatchSnapshot();
          done();
        },
      });
    }, 15000);
  });

  describe('getTimezoneForUnit function', () => {
    it('should throw in case there are NO unit in the DB', done => {
      getTimezoneForUnit({
        crudSdk: iamCrudSdk,
      })('NOT_EXISTING_UNIT_ID').subscribe({
        next() {
          console.error(`${TEST_NAME}Test ERROR`, 'SHOULD NOT SUCCEED');
        },
        error(err) {
          expect(err).toMatchSnapshot();
          done();
        },
      });
    }, 15000);
  });
});
