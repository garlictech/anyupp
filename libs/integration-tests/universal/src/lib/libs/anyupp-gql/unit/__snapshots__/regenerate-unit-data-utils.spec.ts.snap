// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegenerateUnitData Utils tests getTimezoneForUnit function should throw in case there are NO unit in the DB 1`] = `[Error: Unexpected empty value]`;

exports[`RegenerateUnitData Utils tests listUnitProductsForAUnit function should throw in case there are NO unitProducts in the DB 1`] = `"{\\"message\\":\\"There is no product in this unit\\",\\"code\\":\\"ERROR_NO_PRODUCT_IN_UNIT\\"}"`;
